[[36m2023-09-05 09:42:19,536[39m][[34m__main__[39m][[32mINFO[39m] - Instantiating trainer <lightning.pytorch.trainer.Trainer>
[[36m2023-09-05 09:42:19,614[39m][[34mpytorch_lightning.utilities.rank_zero[39m][[32mINFO[39m] - Trainer already configured with model summary callbacks: [<class 'lightning.pytorch.callbacks.rich_model_summary.RichModelSummary'>]. Skipping setting a default `ModelSummary` callback.
[[36m2023-09-05 09:42:19,621[39m][[34mpytorch_lightning.utilities.rank_zero[39m][[32mINFO[39m] - GPU available: True (cuda), used: True
[[36m2023-09-05 09:42:19,621[39m][[34mpytorch_lightning.utilities.rank_zero[39m][[32mINFO[39m] - TPU available: False, using: 0 TPU cores
[[36m2023-09-05 09:42:19,621[39m][[34mpytorch_lightning.utilities.rank_zero[39m][[32mINFO[39m] - IPU available: False, using: 0 IPUs
[[36m2023-09-05 09:42:19,622[39m][[34mpytorch_lightning.utilities.rank_zero[39m][[32mINFO[39m] - HPU available: False, using: 0 HPUs
[[36m2023-09-05 09:42:19,622[39m][[34m__main__[39m][[32mINFO[39m] - Logging hyperparameters!
Running validation before training
LF_generate done
In lf datagen, input shape (153600, 100, 1)
In lf datagen, output shape (153600, 100, 1)
In lf_datamodule.py, self.rho_name pol
LOCAL_RANK: 0 - CUDA_VISIBLE_DEVICES: [2]
/home/shida/anaconda3/envs/lh/lib/python3.9/site-packages/lightning/pytorch/trainer/connectors/data_connector.py:430: PossibleUserWarning: The dataloader, val_dataloader, does not have many workers which may be a bottleneck. Consider increasing the value of the `num_workers` argument` (try 32 which is the number of cpus on this machine) in the `DataLoader` init to improve performance.
  rank_zero_warn(
[[36m2023-09-05 09:42:23,683[39m][[34msrc.models.recurrent.s4[39m][[32mINFO[39m] - S4: Initializing kernel to length 100
┏━━━━━━━━━━━━━━━━━━━━━━━━━━━┳━━━━━━━━━━━━━━━━━━━━━━━━━━━┓
┃[1m      Validate metric      [22m┃[1m       DataLoader 0        [22m┃
┡━━━━━━━━━━━━━━━━━━━━━━━━━━━╇━━━━━━━━━━━━━━━━━━━━━━━━━━━┩
│[36m         val/loss          [39m│[35m    15.279667854309082     [39m│
└───────────────────────────┴───────────────────────────┘
[37mValidation[39m [35m━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━[39m [37m25/25[39m [37m0:00:01 • 0:00:00[39m [37m91.22it/s
[?25h[[36m2023-09-05 09:42:24,226[39m][[34m__main__[39m][[32mINFO[39m] - Starting training!
LOCAL_RANK: 0 - CUDA_VISIBLE_DEVICES: [2]
LF_generate done
In lf datagen, input shape (153600, 100, 1)
In lf datagen, output shape (153600, 100, 1)
┏━━━━┳━━━━━━━━━━━━━━━━━━━━━━━┳━━━━━━━━━━━━━━━┳━━━━━━━━┓
┃[1m    [22m┃[1m Name                  [22m┃[1m Type          [22m┃[1m Params [22m┃
┡━━━━╇━━━━━━━━━━━━━━━━━━━━━━━╇━━━━━━━━━━━━━━━╇━━━━━━━━┩
│ 0  │ net                   │ S4Block       │ 20.7 K │
│ 1  │ net.layer             │ FFTConv       │ 16.5 K │
│ 2  │ net.layer.activation  │ GELU          │      0 │
│ 3  │ net.layer.kernel      │ SSMKernelDPLR │ 16.4 K │
│ 4  │ net.layer.drop        │ Identity      │      0 │
│ 5  │ net.layer.drop_kernel │ Identity      │      0 │
│ 6  │ net.mult_activation   │ Identity      │      0 │
│ 7  │ net.drop              │ Identity      │      0 │
│ 8  │ net.output_linear     │ Sequential    │  4.2 K │
│ 9  │ net.output_linear.0   │ Linear        │  4.2 K │
│ 10 │ net.output_linear.1   │ Hardtanh      │      0 │
│ 11 │ encoder               │ Linear        │    128 │
│ 12 │ encoder.U             │ Linear        │    128 │
│ 13 │ encoder.activation    │ Identity      │      0 │
│ 14 │ decoder               │ Linear        │     65 │
│ 15 │ decoder.U             │ Linear        │     65 │
│ 16 │ decoder.activation    │ Identity      │      0 │
│ 17 │ train_loss            │ MeanMetric    │      0 │
│ 18 │ val_loss              │ MeanMetric    │      0 │
│ 19 │ test_loss             │ MeanMetric    │      0 │
└────┴───────────────────────┴───────────────┴────────┘
[1mTrainable params[22m: 20.9 K
[1mNon-trainable params[22m: 0
[1mTotal params[22m: 20.9 K
[1mTotal estimated model params size (MB)[22m: 0
/home/shida/anaconda3/envs/lh/lib/python3.9/site-packages/lightning/pytorch/trainer/connectors/data_connector.py:430: PossibleUserWarning: The dataloader, train_dataloader, does not have many workers which may be a bottleneck.
Consider increasing the value of the `num_workers` argument` (try 32 which is the number of cpus on this machine) in the `DataLoader` init to improve performance.
  rank_zero_warn(



Epoch 1/999 [35m━━━━╸[90m━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━[39m [37m30/250[39m [37m0:00:00 • 0:00:06[39m [37m37.91it/s[39m [37mv_num: zfk9 train/loss: 0.209 val/loss: 0.481 




Epoch 2/999 [35m━━━━━━━╸[90m━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━[39m [37m48/250[39m [37m0:00:01 • 0:00:06[39m [37m40.16it/s[39m [37mv_num: zfk9 train/loss: 0.08 val/loss: 0.126 



Epoch 3/999 [35m━━╸[90m━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━[39m [37m16/250[39m [37m0:00:00 • 0:00:07[39m [37m38.62it/s[39m [37mv_num: zfk9 train/loss: 0.055 val/loss: 0.021 




Epoch 4/999 [35m━━━━━━━━[90m╺━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━[39m [37m51/250[39m [37m0:00:01 • 0:00:05[39m [37m40.19it/s[39m [37mv_num: zfk9 train/loss: 0.006 val/loss: 0.018 







Epoch 6/999 [35m━━━━━━━━╸[90m━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━[39m [37m55/250[39m [37m0:00:01 • 0:00:05[39m [37m39.60it/s[39m [37mv_num: zfk9 train/loss: 0.005 val/loss: 0.01 


Epoch 6/999 [35m━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━[39m [37m250/250[39m [37m0:00:06 • 0:00:00[39m [37m38.63it/s[39m [37mv_num: zfk9 train/loss: 0.005 val/loss: 0.004 
[?25h[[36m2023-09-05 09:43:15,692[39m][[34m__main__[39m][[32mINFO[39m] - Starting testing!
Stopping threshold reached: val/loss = 0.00435843039304018 < 0.009552148170769215. Signaling Trainer to stop.
LF_generate done
In lf datagen, input shape (153600, 100, 1)
In lf datagen, output shape (153600, 100, 1)
[[36m2023-09-05 09:43:16,930[39m][[34mpytorch_lightning.utilities.rank_zero[39m][[32mINFO[39m] - Restoring states from the checkpoint path at /home/shida/Figure_1/logs/LF_hardtanh_s4_pol/runs/20230905/094212/checkpoints/epoch_006.ckpt
[[36m2023-09-05 09:43:16,949[39m][[34mpytorch_lightning.utilities.rank_zero[39m][[32mINFO[39m] - Loaded model weights from the checkpoint at /home/shida/Figure_1/logs/LF_hardtanh_s4_pol/runs/20230905/094212/checkpoints/epoch_006.ckpt
┏━━━━━━━━━━━━━━━━━━━━━━━━━━━┳━━━━━━━━━━━━━━━━━━━━━━━━━━━┓
┃[1m      Validate metric      [22m┃[1m       DataLoader 0        [22m┃
┡━━━━━━━━━━━━━━━━━━━━━━━━━━━╇━━━━━━━━━━━━━━━━━━━━━━━━━━━┩
│[36m         val/loss          [39m│[35m    0.00435843039304018    [39m│
└───────────────────────────┴───────────────────────────┘
[37mValidation[39m [35m━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━[39m [37m25/25[39m [37m0:00:00 • 0:00:00[39m [37m90.14it/s
[?25h
[?25hLF_generate done
In lf datagen, input shape (153600, 100, 1)
In lf datagen, output shape (153600, 100, 1)
[[36m2023-09-05 09:43:18,782[39m][[34mpytorch_lightning.utilities.rank_zero[39m][[32mINFO[39m] - Restoring states from the checkpoint path at /home/shida/Figure_1/logs/LF_hardtanh_s4_pol/runs/20230905/094212/checkpoints/epoch_006.ckpt
[[36m2023-09-05 09:43:18,802[39m][[34mpytorch_lightning.utilities.rank_zero[39m][[32mINFO[39m] - Loaded model weights from the checkpoint at /home/shida/Figure_1/logs/LF_hardtanh_s4_pol/runs/20230905/094212/checkpoints/epoch_006.ckpt
┏━━━━━━━━━━━━━━━━━━━━━━━━━━━┳━━━━━━━━━━━━━━━━━━━━━━━━━━━┓
┃[1m        Test metric        [22m┃[1m       DataLoader 0        [22m┃
┡━━━━━━━━━━━━━━━━━━━━━━━━━━━╇━━━━━━━━━━━━━━━━━━━━━━━━━━━┩
│[36m         test/loss         [39m│[35m   0.004405083600431681    [39m│
└───────────────────────────┴───────────────────────────┘
[37mTesting[39m [35m━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━[39m [37m25/25[39m [37m0:00:00 • 0:00:00[39m [37m89.88it/s
[?25h[[36m2023-09-05 09:43:19,392[39m][[34m__main__[39m][[32mINFO[39m] - Best ckpt path: /home/shida/Figure_1/logs/LF_hardtanh_s4_pol/runs/20230905/094212/checkpoints/epoch_006.ckpt
[[36m2023-09-05 09:43:19,394[39m][[34msrc.utils.utils[39m][[32mINFO[39m] - Output dir: /home/shida/Figure_1/logs/LF_hardtanh_s4_pol/runs/20230905/094212
[[36m2023-09-05 09:43:19,394[39m][[34msrc.utils.utils[39m][[32mINFO[39m] - Closing wandb!
LOCAL_RANK: 0 - CUDA_VISIBLE_DEVICES: [2]
/home/shida/anaconda3/envs/lh/lib/python3.9/site-packages/lightning/pytorch/trainer/connectors/data_connector.py:430: PossibleUserWarning: The dataloader, test_dataloader, does not have many workers which may be a bottleneck. Consider increasing the value of the `num_workers` argument` (try 32 which is the number of cpus on this machine) in the `DataLoader` init to improve performance.
  rank_zero_warn(